object-search/pods/craft.yml

```diff
@@ -0,0 +1,56 @@
+!CompoundExecutor
+components:
+  - !ImageReader
+    with:
+      from_bytes: True
+    metas:
+      name: img_read
+  - !ImageNormalizer
+    with:
+      target_size: 96
+      img_mean: [0.485, 0.456, 0.406]
+      img_std: [0.229, 0.224, 0.225]
+      channel_axis: -1
+    metas:
+      name: img_norm
+  - !TorchObjectDetectionSegmenter
+    with:
+      channel_axis: -1
+      confidence_threshold: 0.98
+    metas:
+      name: img_detect
+  - !ImageResizer
+    with:
+      target_size: 96
+    metas:
+      name: img_resize
+metas:
+  name: compound_crafter
+requests:
+  on:
+    IndexRequest:
+      - !CraftDriver
+        with:
+          executor: img_read
+      - !CraftDriver
+        with:
+          executor: img_norm
+      - !CraftDriver
```

Comment:
TorchObjectDetection is a Segmenter, therefore a SegmentDriver should be driving it. I suggest to split these executors in at least 2 steps (load and normalize) and then (detect)

Reply:
Hey Joan, I am getting this after I changed `craftdriver` to `segmentdriver`. Can this be due to TorchObjectDetectionSegmenter confidence_threshold: 0.98

```
crafter@37895[W]:doc 1 at level 0 gives no chunk
crafter@37895[W]:doc 2 at level 0 gives no chunk
```

Reply:
Yes, TorchObjectDetection may not detect anything, for testing you can keep a really low threshold.

Reply:
Keeping threshold to 0.1 detects objects in some images but then throws error about different shape of cropped images while passing through the next encoder. I have used resize as shown below.

```
  - !TorchObjectDetectionSegmenter
    with:
      channel_axis: -1
      confidence_threshold: 0.1
    metas:
      name: img_detect
  - !ImageResizer
    with:
      target_size: 96
    metas:
      name: img_resize
```

Also tried

```
  - !TorchObjectDetectionSegmenter
    with:
      target_size: (96, 96)
      channel_axis: -1
      confidence_threshold: 0.1
    metas:
      name: img_detect
  - !ImageResizer
    with:
      target_size: 96
    metas:
      name: img_resize
```

vec_idx@38553[I]:received "index" from gateway▸crafter▸encoder▸⚐
        vec_idx@38553[W]:these bad docs can not be added: [(2450309498, 1), (3283606135, 1)]
        vec_idx@38553[W]:these bad docs can not be added: [(537664483, 4), (1729837641, 4), (3884707008, 4), (1685125731, 4), (1960091019, 4)]
        vec_idx@38553[W]:these bad docs can not be added: [(2830395716, 6), (923790454, 6), (1953433588, 6)]
        vec_idx@38553[W]:these bad docs can not be added: [(1, 0), (2, 0), (3, 0), (4, 0), (5, 0), (6, 0)]
       join_all@38555[I]:received "index" from gateway▸crafter▸encoder▸vec_idx▸⚐
       join_all@38555[I]:collected 2/2 parts of IndexRequest
        gateway@38544[E]:callback() may not work properly due to the bad response: encoder(ImageKerasEncoder) throws ValueError('all input arrays must have the same shape')

Reply:
I think u did not use ImageResizer with the right granularity level, u applied ImageResizer to the Original image and is not what is being indexed. First,, let's divide this crafter in at least 2 different Pods